gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/iceberg/IcebergDataset.java

```diff
@@ -185,31 +185,28 @@ protected Map<Path, FileStatus> getFilePathsToFileStatus() throws IOException {
 
     for (String pathString : pathsToCopy) {
       Path path = new Path(pathString);
-      result.put(path, this.fs.getFileStatus(path));
+      result.put(path, this.sourceFs.getFileStatus(path));
     }
     return result;
   }
 
-  DatasetDescriptor getSourceDataset() {
-    return getDatasetDescriptor(sourceMetastoreURI);
+  DatasetDescriptor getSourceDataset(FileSystem sourceFs) {
+    return getDatasetDescriptor(sourceMetastoreURI, sourceFs);
   }
 
-  DatasetDescriptor getDestinationDataset() {
-    return getDatasetDescriptor(targetMetastoreURI);
+  DatasetDescriptor getDestinationDataset(FileSystem targetFs) {
```

Comment:
shouldn't these either be `@VisibleForTesting` or `protected`?

Reply:
good callout.... made them protected and updated the file!